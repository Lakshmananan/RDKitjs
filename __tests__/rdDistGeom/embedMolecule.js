'use strict';

const RDKit = require('../..');

beforeEach(() => RDKit.load());

test('manual method', () => {
  const smiles = 'COCO';
  const mol = RDKit.Mol.fromSmiles(smiles);
  mol.addHs();
  mol.embedMolecule();
  expect(mol.toMolBlock()).toMatchSnapshot();
});

test('ETDG method', () => {
  const smiles = 'COCO';
  const mol = RDKit.Mol.fromSmiles(smiles);
  mol.addHs();
  mol.embedMolecule({ method: 'ETDG' });
  expect(mol.toMolBlock()).toMatchSnapshot();
});

test('ETKDG method', () => {
  const smiles = 'COCO';
  const mol = RDKit.Mol.fromSmiles(smiles);
  mol.addHs();
  mol.embedMolecule({ method: 'ETKDG' });
  expect(mol.toMolBlock()).toMatchSnapshot();
});

test('KDG method', () => {
  const smiles = 'COCO';
  const mol = RDKit.Mol.fromSmiles(smiles);
  mol.addHs();
  mol.embedMolecule({ method: 'KDG' });
  expect(mol.toMolBlock()).toMatchSnapshot();
});

test('wrong method', () => {
  const smiles = 'COCO';
  const mol = RDKit.Mol.fromSmiles(smiles);
  mol.addHs();
  expect(() => mol.embedMolecule({ method: 'wrong' })).toThrow(
    /unknown embed method: wrong/
  );
});
