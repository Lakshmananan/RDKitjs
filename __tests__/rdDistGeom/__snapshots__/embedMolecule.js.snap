// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ETDG method 1`] = `
"
     RDKit          3D

 10  9  0  0  0  0  0  0  0  0999 V2000
   -1.2103   -0.1870   -0.2321 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.2935   -0.2915    0.8069 O   0  0  0  0  0  0  0  0  0  0  0  0
    0.8772    0.3952    0.5270 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.5126   -0.0793   -0.6182 O   0  0  0  0  0  0  0  0  0  0  0  0
   -0.8633   -0.5807   -1.1923 H   0  0  0  0  0  0  0  0  0  0  0  0
   -1.5497    0.8544   -0.3913 H   0  0  0  0  0  0  0  0  0  0  0  0
   -2.1148   -0.7811    0.0740 H   0  0  0  0  0  0  0  0  0  0  0  0
    1.5595    0.1960    1.3994 H   0  0  0  0  0  0  0  0  0  0  0  0
    0.7819    1.4778    0.4595 H   0  0  0  0  0  0  0  0  0  0  0  0
    1.3004   -1.0037   -0.8329 H   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0
  2  3  1  0
  3  4  1  0
  1  5  1  0
  1  6  1  0
  1  7  1  0
  3  8  1  0
  3  9  1  0
  4 10  1  0
M  END
"
`;

exports[`ETKDG method 1`] = `
"
     RDKit          3D

 10  9  0  0  0  0  0  0  0  0999 V2000
   -1.3155   -0.0256    0.7188 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.0184   -0.3414    1.0290 O   0  0  0  0  0  0  0  0  0  0  0  0
    0.7437   -0.0468   -0.1344 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.0953   -0.3129    0.0258 O   0  0  0  0  0  0  0  0  0  0  0  0
   -1.9031   -0.2526    1.6281 H   0  0  0  0  0  0  0  0  0  0  0  0
   -1.3985    1.0646    0.4487 H   0  0  0  0  0  0  0  0  0  0  0  0
   -1.7031   -0.6526   -0.1160 H   0  0  0  0  0  0  0  0  0  0  0  0
    0.5276    1.0135   -0.3699 H   0  0  0  0  0  0  0  0  0  0  0  0
    0.2717   -0.6518   -0.9375 H   0  0  0  0  0  0  0  0  0  0  0  0
    2.6637    0.2055   -0.5652 H   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0
  2  3  1  0
  3  4  1  0
  1  5  1  0
  1  6  1  0
  1  7  1  0
  3  8  1  0
  3  9  1  0
  4 10  1  0
M  END
"
`;

exports[`KDG method 1`] = `
"
     RDKit          3D

 10  9  0  0  0  0  0  0  0  0999 V2000
   -1.3674    0.0596    0.0333 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.2905    0.6254    0.6669 O   0  0  0  0  0  0  0  0  0  0  0  0
    0.9338    0.2635    0.2095 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.9690   -0.8362   -0.6189 O   0  0  0  0  0  0  0  0  0  0  0  0
   -1.3583    0.1201   -1.0584 H   0  0  0  0  0  0  0  0  0  0  0  0
   -2.2809    0.6180    0.3853 H   0  0  0  0  0  0  0  0  0  0  0  0
   -1.5419   -0.9985    0.3154 H   0  0  0  0  0  0  0  0  0  0  0  0
    1.3911    1.1365   -0.3221 H   0  0  0  0  0  0  0  0  0  0  0  0
    1.6019    0.0955    1.0960 H   0  0  0  0  0  0  0  0  0  0  0  0
    1.9432   -1.0838   -0.7069 H   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0
  2  3  1  0
  3  4  1  0
  1  5  1  0
  1  6  1  0
  1  7  1  0
  3  8  1  0
  3  9  1  0
  4 10  1  0
M  END
"
`;

exports[`manual method 1`] = `
"
     RDKit          3D

 10  9  0  0  0  0  0  0  0  0999 V2000
   -1.2659    0.1284    0.0614 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.4608   -0.8387   -0.5524 O   0  0  0  0  0  0  0  0  0  0  0  0
    0.8648   -0.5232   -0.2402 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.9912    0.8457   -0.1939 O   0  0  0  0  0  0  0  0  0  0  0  0
   -2.2681   -0.3255    0.2535 H   0  0  0  0  0  0  0  0  0  0  0  0
   -1.3620    1.0359   -0.5537 H   0  0  0  0  0  0  0  0  0  0  0  0
   -0.7515    0.4217    1.0086 H   0  0  0  0  0  0  0  0  0  0  0  0
    1.1004   -0.9653    0.7410 H   0  0  0  0  0  0  0  0  0  0  0  0
    1.5257   -0.9666   -0.9939 H   0  0  0  0  0  0  0  0  0  0  0  0
    1.6263    1.1876    0.4695 H   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0
  2  3  1  0
  3  4  1  0
  1  5  1  0
  1  6  1  0
  1  7  1  0
  3  8  1  0
  3  9  1  0
  4 10  1  0
M  END
"
`;
